src/object/srv_obj.c

```diff
@@ -1744,8 +1755,12 @@ obj_get_iods_offs_by_oid(daos_unit_oid_t uoid, struct obj_iod_array *iod_array,
 		oiod = &iod_array->oia_oiods[i];
 		if (iod_parent->iod_type == DAOS_IOD_ARRAY) {
 			siod = obj_shard_iod_get(oiod, local_tgt);
-			if (siod == NULL)
+			if (siod == NULL) {
+				/* will be skipped in vos_io akey_update() */
+				(*iods)[idx].iod_nr = 0;
+				idx++;
 				continue;
```

Comment:
Just curious, why can not we do the same thing for single value to skip the bulk transfer? or I miss sth?

Reply:
it actually is same as singv that will skip the akey update (by akey_update(), current master code already with the code to skip array's akey update for that case), then need not do bulk transferring in update_end().

Reply:
I re-checked the code, the original code to skip/bypass one akey's update is actually incorrect and possibly cause space leak. I have changed that part of code by remove the akey/iod before calling into VOS, that can address @jolivier23 's concern as well.